tests/buffer/buffer_constructors_common.h

```diff
@@ -27,6 +27,42 @@
 namespace buffer_constructors_common {
 using namespace sycl_cts;
 
+template <typename Acc>
+struct iota_kernel {
+  Acc acc1, acc2;
+  void operator()(sycl::id<1> id) const {
+    acc1[id] = static_cast<typename Acc::value_type>(id[0]);
+    acc2[id] = static_cast<typename Acc::value_type>(id[0]);
+  }
+};
+
+inline std::stringstream alloc_log;
+
+template <typename T>
+class logging_alloc {
+ public:
+  typedef T value_type;
+  typedef T *pointer;
+  typedef size_t size_type;
+
+  T *allocate(size_t n) {
+    alloc_log << "Allocating " << n << " bytes of storage.\n";
+    T *mem = static_cast<T *>(malloc(sizeof(T) * n));
+    if (mem != nullptr)
+      return mem;
+    else {
+      alloc_log << "Failed!\n";
+      throw std::bad_alloc();
+    }
+  }
+
+  void deallocate(T *p, size_t n) {
+    alloc_log << "Deallocating " << n << " bytes of storage at " << p << "\n";
+    free(p);
+    return;
+  }
```

Comment:
```suggestion
  }
```
I suppose return is not necessary here. Right?